core/context/server_context.go

```diff
@@ -0,0 +1,28 @@
+package context
+
+import (
+	"fmt"
+
+	cmtcfg "github.com/cometbft/cometbft/config"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"cosmossdk.io/log"
+)
+
+const ServerContextKey ContextKey = "server-ctx"
+
+type ServerContext interface {
+	GetLogger() log.Logger
+	GetViper() *viper.Viper
+	GetConfig() *cmtcfg.Config
+}
+
+func GetServerContextFromCmd(cmd *cobra.Command) ServerContext {
+	if v := cmd.Context().Value(ServerContextKey); v != nil {
+		fmt.Println("serverCtxPtr", v)
+		serverCtxPtr := v.(ServerContext)
+		return serverCtxPtr
+	}
+	return nil
+}
```

Comment:
this is specific to server/v2 it can live there. Ideally there is no comet deep in server/v2 as well

Reply:
Make sense, I will make `*cmtcfg.Config` as an interface